### PR TITLE
feat(switch): add vertical switch function

### PR DESCRIPTION
**Pull request by lizhaoming634**
A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
Thank you! 

Please add an example and test as well.

**Comment by kisvegabor** [comment]
Please also create [a similar API](https://github.com/lvgl/lvgl/blob/master/src/widgets/bar/lv_bar.h#L37-L41) for the switch.

**Comment by kisvegabor** [comment]
Great!

It just needs to be made complete by:
- Mention it in the docs [here](https://github.com/lvgl/lvgl/blob/master/docs/widgets/switch.rst)
- Add an example similar to [this](https://github.com/lvgl/lvgl/blob/master/examples/widgets/switch/lv_example_switch_1.c) Show a few switches with different sizes and orientation. 
- Convert the example to a test with screenshot compare. Read more [here](https://github.com/lvgl/lvgl/blob/master/tests/README.md)

**Comment by 100ask** [comment]
I suggest adjusting the direction automatically based on width and height, similar to [slider](https://github.com/lvgl/lvgl/blob/master/src/widgets/slider/lv_slider.c#L305), for example:

```c
static bool is_switch_horizontal(lv_obj_t * obj)
{
    return lv_obj_get_width(obj) >= lv_obj_get_height(obj);
}

const bool is_horizontal = is_switch_horizontal(obj);
```

**Comment by lizhaoming19980614** [comment]
no previous prototype for ‘lv_example_switch_2’ [-Werror=missing-prototypes] ，How should I handle these errors? @kisvegabor @FASTSHIFT

File: src/widgets/switch/lv_switch.c [hunk 1]
```diff
@@ -80,6 +80,31 @@ lv_obj_t * lv_switch_create(lv_obj_t * parent)
     return obj;
 }
 
+/*=====================
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.c:84]: remove

File: src/widgets/switch/lv_switch.c [hunk 2]
```diff
@@ -80,6 +80,31 @@ lv_obj_t * lv_switch_create(lv_obj_t * parent)
...
+void lv_switch_set_orientation(lv_obj_t * obj, lv_switch_orientation_t orientation)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+    lv_switch_t * sw = (lv_switch_t *)obj;
+
+    sw->orientation = orientation;
+    lv_obj_invalidate(obj);
+}
+
+/*=====================
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.c:97]: remove

File: src/widgets/switch/lv_switch.c [hunk 3]
```diff
@@ -80,6 +80,31 @@ lv_obj_t * lv_switch_create(lv_obj_t * parent)
     return obj;
 }
 
+/*=====================
+ * Setter functions
+ *====================*/
+
+void lv_switch_set_orientation(lv_obj_t * obj, lv_switch_orientation_t orientation)
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.c:88]: lv_switch_set_direction

File: src/widgets/switch/lv_switch.h [hunk 4]
```diff
@@ -28,6 +28,16 @@ extern "C" {
 
 LV_ATTRIBUTE_EXTERN_DATA extern const lv_obj_class_t lv_switch_class;
 
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef enum {
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.h:35]: It is recommended to use the existing enumeration `lv_dir_t`
`LV_DIR_NONE` as auto mode.

File: src/widgets/switch/lv_switch.c (hunk 3 above)
Comment (FASTSHIFT) [reply src/widgets/switch/lv_switch.c:88]: ignore it.

File: src/widgets/switch/lv_switch.h (hunk 4 above)
Comment (FASTSHIFT) [reply src/widgets/switch/lv_switch.h:35]: ignore it.

File: src/widgets/switch/lv_switch.h [hunk 5]
```diff
@@ -39,6 +49,28 @@ LV_ATTRIBUTE_EXTERN_DATA extern const lv_obj_class_t lv_switch_class;
  */
 lv_obj_t * lv_switch_create(lv_obj_t * parent);
 
+/*=====================
+ * Setter functions
+ *====================*/
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.h:54]: remove

File: src/widgets/switch/lv_switch.h [hunk 6]
```diff
@@ -39,6 +49,28 @@ LV_ATTRIBUTE_EXTERN_DATA extern const lv_obj_class_t lv_switch_class;
...
+/**
+ * Set the orientation of switch.
+ * @param obj           pointer to switch object
+ * @param orientation   switch orientation from `lv_switch_orientation_t`
+ */
+void lv_switch_set_orientation(lv_obj_t * obj, lv_switch_orientation_t orientation);
+
+/*=====================
+ * Getter functions
+ *====================*/
```
Comment (FASTSHIFT) [inline src/widgets/switch/lv_switch.h:65]: remove

Comment (100ask) [reply src/widgets/switch/lv_switch.h:65]: Hi, I don't quite understand what you mean. I don't think these should be removed, refer to: https://github.com/lvgl/lvgl/blob/master/src/widgets/bar/lv_bar.h#L37-L41

Comment (FASTSHIFT) [reply src/widgets/switch/lv_switch.h:65]: > Hi, I don't quite understand what you mean. I don't think these should be removed, refer to: https://github.com/lvgl/lvgl/blob/master/src/widgets/bar/lv_bar.h#L37-L41

OK

**Review by FASTSHIFT — DISMISSED** [review]

**Review by kisvegabor — DISMISSED** [review]

**Comment by kisvegabor** [comment]
I've just added a test realized that the off state should be on the bottom. Please check if you agree with this change.

**Comment by lizhaoming19980614** [comment]
> I've just added a test realized that the off state should be on the bottom. Please check if you agree with this change.我刚刚添加了一个测试，意识到 off 状态应该在底部。请检查您是否同意此更改。

yes, I agree this change.

**Review by kisvegabor — APPROVED** [review]

**Review by 100ask — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**